### ISSUE TITLE
How to implement this in RPG Maker Mv?

### ISSUE THREAD

**Issue by Jogo78** [pixijs/pixi-lights#36]
Hello,
I'm implemented projection spine in RPG Maker Mv,
Now, I'm want to implement lights too, but these files.js,
Isn't compatible...

There is this forum but it's not work :
[https://forums.rpgmakerweb.com/index.php?threads/lights-in-pixi-js.81370/](url)

Some body to help me??

**Comment by ivanpopelyshev** [comment]
@djmisterjon knows how to do it.

Bear in mind that both projections and lights are experimental plugins and you have to be aware of how their work if you want to use them for something more than simple demos, especially if you want to use them together.

Maybe you wont be able to use them at the same time, maybe you have to wait when pixi-display supports multi-texture sprites, or something like that.

Are you sure you want normal maps in your game and not something simpler like [here](https://pixijs.io/examples/#/layers/lighting.js) ?

Also user `anisoft` at pixijs slack knows how to use different types of lighting in RMMV. Do you want an invite there?

**Comment by Jogo78** [comment]
Yes I'm interested.
Thanks for the invitation.

**Comment by ivanpopelyshev** [comment]
give me your email here, or write me to ivan.popelyshev@gmail.com

**Comment by Jogo78** [comment]
I'm send message in your mail.
Thanks and sorry for my english.

@djmisterjon speak french, he can explain me no?
How to contact him?

**Comment by Copywright** [comment]
Also interested in this issue, let me know if anyone finds a solution...

**Comment by ivanpopelyshev** [comment]
The plugin is experimental and people who are using it with RMMV.. not actually using it with vanilla RMMV, they already have very serious experience with plugins and their games are heavy-modded.

If you are sure, I can send you an invite to pixijs slack :)

**Comment by Jogo78** [comment]
I managed to make the sprite black but not enlightened, I'm not put normal, you think that's the problem ????

**Comment by ivanpopelyshev** [comment]
Sorry guys, I don't have time to explain everything, I'm working on v5 and moving plugins to it, including this one. 

so far @djmisterjon  is the only one who made it work.

**Comment by Jogo78** [comment]
It's this without normal no light :D

**Comment by ivanpopelyshev** [comment]
1. make sure updateStage() is called from your app, from somewhere. Otherwise layers wont work at all.
2. make sure that lightLayer has _activeChildren with your lights inside.

**Comment by ltngames** [comment]
I've never implemented pixi-lights into RPGMaker MV but I am using pixi-layers and the basic light example from pixi examples to create a lighting system of my own for RM game. My guess is pixi-lights will not be much different than doing things this way.

Take a look at examples below, I can't show you the exact way I'm doing it without confusing the crap out of you so I moved my code around to show you it a bit easier, I never tested the code, it's just an extraction of my code placed into one example to quickly show you. In my real code I use a class called LightManager to add the layers to Spriteset_Map and to add lights to the lighting layer.


The first thing you need to do is make sure Stage is using pixi-layers. 
```js
function Stage () {
  this.initialize.apply(this, arguments)
}

// Use PIXI Layers
Stage.prototype = Object.create(PIXI.display.Stage.prototype)
Stage.prototype.constructor = Stage

Stage.prototype.initialize = function () {
  PIXI.display.Stage.call(this)

  // The interactive flag causes a memory leak.
  this.interactive = false
}
```

Then add a lighting layer and a blend sprite to the map's spriteset
```js
// Sorry about bad var names lol

  // Create lighting layer's that make the lights shine!
  const _lightingLayer = new PIXI.display.Layer()

  _lightingLayer.on('display', (element) => {
    element.blendMode = PIXI.BLEND_MODES.ADD
  })
  _lightingLayer.useRenderTexture = true
  _lightingLayer.clearColor = _ambience

  const _lightBlendLayer = new PIXI.Sprite(_lightingLayer.getRenderTexture())
  _lightBlendLayer.blendMode = PIXI.BLEND_MODES.MULTIPLY

// Alias createUpperLayer, create lights and add the layers and lights
const createUpperLayer = Spriteset_Map.prototype.createUpperLayer
Spriteset_Map.prototype.createUpperLayer = function () {
  createUpperLayer.call(this)
 this.addChild(_lightingLayer)
 this.addChild(_lightBlendLayer)

// Create two sprites, one is the light and another to hold the light, it's the only way this will work.
  const _lightContainer = new Sprite()
  const _lightSprite = new Sprite()

 /*
 Give lightSprite a light image for it's bitmap, you may also use Bitmap.drawCircle(x, y, radius, color) or PIXI.Graphics, or whatever you want really.
 */
    _lightSprite.bitmap = ImageManager.loadBitmap('img/lights/', filename, 0, true)
 _lightSprite.parentLayer = _lightingLayer

 // Add lightSprite to the container
 _lightContainer.addChild(lightSprite)

 // Add the lightContainer to the lightingLayer
 _lightingLayer.addChild(_lightContainer)
}
```

**Comment by ivanpopelyshev** [comment]
Yeah, i dont know why people go for lights with normals before they try simple solution. Of course its possible even without pixi-display, its just an efficient way to code multi-layered world.
